wdtk-datamodel/src/main/java/org/wikidata/wdtk/datamodel/implementation/StatementImpl.java

```diff
@@ -74,6 +74,36 @@
 
 	private List<SnakGroup> qualifiersGroups;
 
+	/**
+	 * Checks if the statement ID consists of a first part
+	 * with the entity id of the item the statement belongs to, the separator $, plus
+	 * a random hash of the form
+	 * /^\{?[A-Z\d]{8}-[A-Z\d]{4}-[A-Z\d]{4}-[A-Z\d]{4}-[A-Z\d]{12}\}?\z/
+	 *
+	 * @param statementID
+	 * @return true if the statement ID is of the valid format
+	 */
+	public static boolean validateStatementID(String statementID)
+	{
+		if(statementID.equals(""))
+			return true;
+
+		String separator = "\\$";
+		String statementFormat ="^\\{?[A-Za-z0-9]{8}-[A-Za-z0-9]{4}-[A-Za-z0-9]{4}-[A-Za-z0-9]{4}-[A-Za-z0-9]{12}\\}?$";
```

Comment:
We will validate millions of IDs. I belive it would be better to precompile the pattern using `Pattern.compile` and store it in a `final static` class field.